### PR TITLE
NMS-6584: Upgrade to Quartz 2.2.X

### PR DESCRIPTION
**Pull request by soleger**
Upgrade Quartz to a modern version so that we have a version that's compatible with Spring 4.1.
- JIRA: http://issues.opennms.org/browse/NMS-6584
- Bamboo: http://bamboo.internal.opennms.com:8085/browse/OPENNMS-ONMS856

### CODE REVIEW

File: dependencies/quartz/pom.xml [hunk 1]
```diff
@@ -12,22 +12,13 @@
...
   </dependencies>
-  <repositories>
-    <repository>
-      <snapshots><enabled>false</enabled></snapshots>
-      <releases><enabled>true</enabled></releases>
-      <id>jboss-repo</id>
-      <name>JBoss Repository</name>
-      <url>http://maven.opennms.org/content/repositories/jboss-repo/</url>
-    </repository>
-  </repositories>
```
Comment (mvrueden) [inline dependencies/quartz/pom.xml]: 👍

**Comment by mvrueden** [comment]
Looks good to me. The only question I have is, is it really required to use the *Impl classes? This looks kinda weird to me. Or is this just the way Quartz is supposed to be used?

**Comment by soleger** [comment]
Quartz 2 changed their classes into interface/impl pairs so using the *Impl classes directly is the right thing to do afaik.

Note that this branch is failing in Bamboo due to some change that isn't correct inside the Poller Spring context. I just wanted to add a PR for it so that I don't forget about the work. It would be nice to get this done in the 19 timeframe. It's one of the last blockers for upgrading to Spring 4.1.

**Comment by j-white** [comment]
I don't think we'll be able to upgrade Quartz without upgrading Spring at the same time. The tests are currently failing with the following trace:

```
java.lang.NullPointerException: null
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.springframework.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:209)
	at org.springframework.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:194)
	at org.springframework.scheduling.quartz.SimpleTriggerFactoryBean.afterPropertiesSet(SimpleTriggerFactoryBean.java:271)
	at org.opennms.netmgt.poller.remote.PolledServiceTrigger.<init>(PolledServiceTrigger.java:55)
```

**Comment by mvrueden** [comment]
Hey Ulf, please check the OCA.

**Comment by j-white** [comment]
Closing work in progress, feel free to re-open when this is complete.

**Pull request closed without merging**